django/db/backends/postgresql/introspection.py

```diff
@@ -22,15 +23,45 @@ class DatabaseIntrospection(BaseDatabaseIntrospection):
         700: 'FloatField',
         701: 'FloatField',
         869: 'GenericIPAddressField',
+        1000: ('django.contrib.postgresql.fields.ArrayField',
```

Comment:
For example, https://github.com/adamchainz/django-mysql offers custom fields without providing a custom database backend. I guess there's probably not introspection support but maybe @adamchainz has some ideas about this.